File: src/main/java/hu/frontrider/arcana/content/items/ItemBase.java
package hu.frontrider.arcana.content.items;

import net.minecraft.item.Item;
import net.minecraft.util.ResourceLocation;

import static hu.frontrider.arcana.ThaumicArcana.INSTANCE;
import static hu.frontrider.arcana.ThaumicArcana.MODID;

public class ItemBase extends Item {

    ItemBase(ResourceLocation resourceLocation) {
        setRegistryName(resourceLocation);
        setUnlocalizedName(MODID + "." + resourceLocation.getResourcePath());
        setCreativeTab(INSTANCE.getTABARCANA());
    }
}

File: src/main/java/hu/frontrider/arcana/content/items/ItemBase.kt
package hu.frontrider.arcana.content.items

import net.minecraft.item.Item
import net.minecraft.util.ResourceLocation

import hu.frontrider.arcana.ThaumicArcana.MODID
import hu.frontrider.arcana.ThaumicArcana.TABARCANA

open class ItemBase : Item {

    internal constructor(name: String) {
        setRegistryName(MODID, name)
        unlocalizedName = "$MODID.$name"
        setCreativeTab(TABARCANA)
    }

    internal constructor(resourceLocation: ResourceLocation) {
        registryName = resourceLocation
        unlocalizedName = MODID + "." + resourceLocation.resourcePath
        setCreativeTab(TABARCANA)
    }
}
